Sources/MongoQueue/KnownType.swift

```diff
@@ -112,6 +125,8 @@ internal struct KnownType {
             let onFailure = try await metadata.onExecutionFailure(failureContext: failureContext)
             
             func applyRemoval(_ removal: TaskRemovalAction) async throws {
+                queue.jobsRemoved.increment()
```

Comment:
Would we wanna move this to after the removal of the job and not before? Could give us an inaccurate signal if something failed inserting/upserting.